.github/workflows/openstack-ubuntu-20.yml

```diff
@@ -0,0 +1,632 @@
+
+        - name: Export VM_IP to env
+          env:
+            VM_IP: ${{ steps.VM_IP.outputs.VM_IP }}
+          run: echo "VM_IP is ${{ steps.VM_IP.outputs.VM_IP }}"
+        - name: Monitor Elevate for Final Reboot
+          uses: appleboy/ssh-action@v1.1.0
+          with:
+            host: ${{ steps.VM_IP.outputs.VM_IP }}
+            username: 'root'
+            key: ${{ secrets.SSH_PRIVATE_KEY }}
+            port: '22'
+            timeout: 45m
+            command_timeout: 35m
+            script: |
+              /scripts/status_marker 5
+              tail -n40 -F /var/log/elevate-cpanel.log &
+              sleep 2.5
+              SKIP_PID_CHECK=1 REBOOT_STRING="Doing final reboot" RETVAL=1 /scripts/reboot_watch
+
+    wait_for_final_reboot:
+      runs-on: self-hosted
+      needs: watch_for_final_reboot
+      outputs:
+        VM_IP: ${{ steps.VM_IP.outputs.VM_IP }}
+      steps:
+        - name: Download VM IP
+          uses: actions/download-artifact@v4.1.8
+          with:
+            name: ${{ github.run_id }}-ubuntu20-vm_ip
+            path: ${{ github.workspace }}/
+
+        - name: Get VM IP from Artifact
+          id: VM_IP
+          run: |
+            echo "VM_IP=$(cat ${{ github.run_id }}-vm_ip)" >> "$GITHUB_OUTPUT"
+            cat ${{ github.run_id }}-vm_ip > VM_IP
+
+        - name: Export VM_IP to env
+          env:
+            VM_IP: ${{ steps.VM_IP.outputs.VM_IP }}
+          run: echo "VM_IP is ${{ steps.VM_IP.outputs.VM_IP }}"
+        - name: Wait For VM to Come Back From Final Reboot
+          working-directory: "./.github/workflows/openstack/"
+          run: |
+            ./ssh_retry ${{ steps.VM_IP.outputs.VM_IP }}
+
+    verify_upgraded_os:
+      runs-on: self-hosted
+      needs: wait_for_final_reboot
+      outputs:
+        VM_IP: ${{ steps.VM_IP.outputs.VM_IP }}
+      steps:
+        - name: Download VM IP
+          uses: actions/download-artifact@v4.1.8
+          with:
+            name: ${{ github.run_id }}-ubuntu20-vm_ip
+            path: ${{ github.workspace }}/
+
+        - name: Get VM IP from Artifact
+          id: VM_IP
+          run: |
+            echo "VM_IP=$(cat ${{ github.run_id }}-vm_ip)" >> "$GITHUB_OUTPUT"
+            cat ${{ github.run_id }}-vm_ip > VM_IP
+
+        - name: Export VM_IP to env
+          env:
+            VM_IP: ${{ steps.VM_IP.outputs.VM_IP }}
+          run: echo "VM_IP is ${{ steps.VM_IP.outputs.VM_IP }}"
+        - name: Verify End Result Integration Tests
+          uses: appleboy/ssh-action@v1.1.0
+          with:
+            host: ${{ steps.VM_IP.outputs.VM_IP }}
+            username: 'root'
+            key: ${{ secrets.SSH_PRIVATE_KEY }}
+            port: '22'
+            timeout: 5m
+            command_timeout: 1m
+            script: |
+              REPODIR=$(echo ${{ github.repository }} | cut -d / -f2)
+              /usr/local/cpanel/3rdparty/bin/prove -lvm /opt/${REPODIR}/t/integration/ubuntu7-to-almalinux8/*.t
```

Comment:
FIX: This should be a new directory `ubuntu20-to-ununtu22` and a new test similar to what is done in `t/integration/centos7-to-almalinux8/complete.t` will need to be created to verify that the server is `Ubuntu 22.04` after elevate completes.